Core/PrimaryGenerator.cpp

```diff
@@ -197,14 +205,20 @@ void PrimaryGenerator::GenerateEvtSmallChamber(G4Event* event, const G4double ef
   }
   else if (evtFractions[0] + evtFractions[1] + evtFractions[2] + evtFractions[3] > random) // Direct 3G
   {
-    event->AddPrimaryVertex(GenerateThreeGammaVertex(vtxPosition, T0, 
+    event->AddPrimaryVertex(GenerateThreeGammaVertex( MaterialExtension::DecayChannel::Direct, vtxPosition, T0, 
                             material->GetLifetime(random - evtFractions[0] - evtFractions[1] - evtFractions[2], MaterialExtension::DecayChannel::Direct)));
   }
-  else // oPs 3G
+  else if (evtFractions[0] + evtFractions[1] + evtFractions[2] + evtFractions[3] + evtFractions[4] > random) // oPs 3G
   {
-    event->AddPrimaryVertex( GenerateThreeGammaVertex(vtxPosition, T0, 
+    event->AddPrimaryVertex( GenerateThreeGammaVertex( MaterialExtension::DecayChannel::Ortho3G, vtxPosition, T0, 
                             material->GetLifetime(random - evtFractions[1] - evtFractions[2] - evtFractions[3] - evtFractions[4], MaterialExtension::DecayChannel::Ortho3G)));
```

Comment:
Oh...Now i see that I made a mistake there. It should be 
random - evtFractions[0] - evtFractions[1] - evtFractions[2] - evtFractions[3]
because in that sense (random - evtFractions[1] - evtFractions[2] - evtFractions[3] - evtFractions[4]) can be negative
and in general it should not be. Could you correct that here?